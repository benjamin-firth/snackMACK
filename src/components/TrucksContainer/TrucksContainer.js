import React, { Component } from 'react';
import { connect } from 'react-redux';
import './TrucksContainer.scss';
import Header from '../Header/Header';
import Truck from '../Truck/Truck';
import PotentialLocations from '../PotentialLocations/PotentialLocations';
import TruckMap from '../Map/Map';
import { fetchFoodTrucks } from '../../utils/apiCalls';
import { setTrucks, togglePotentialLocation } from '../../actions';

export class TrucksContainer extends Component {
  constructor() {
    super();
    this.state = {
      showMap: true
    }
  }

  componentDidMount() {
    fetchFoodTrucks(this.props.city.city)
      .then(data => {
        const vendors = data.vendors;
        const vendorKeys = Object.keys(vendors);
        const nearbyTrucks = vendorKeys.map(key => {
          return { 
            name: vendors[key].name, 
            desc: vendors[key].description,
            location: vendors[key].last,
            image: vendors[key].images,
            email: vendors[key].url,
            isPotentialLocation: false
          }
        });
        this.props.setTrucks(nearbyTrucks);
      })
  }

  toggleMap = () => {
    this.setState({showMap: !this.state.showMap})
  }

  render() {
    const Trucks = this.props.allTrucks.map(truck => {
      return (
        <Truck 
          truck={truck}
          key={truck.name}
        />) 
    })

    return (
      <main>
        <Header />
        <div className='mainpage-container'>
          <div className='sidebar'>
            <button onClick={this.toggleMap}>{this.state.showMap? 'VIEW POTENTIAL LOCATIONS' : 'VIEW MAP OF TRUCKS'}</button>
            <div className='inner-scroll-container'>
              {Trucks}
            </div>
          </div>
          {this.state.showMap? (
            <TruckMap 
              lat={this.props.city.lat}
              long={this.props.city.long}
              trucks={this.props.allTrucks}
              togglePotentialLocation={this.props.togglePotentialLocation}
              key={this.props.city}
              />
          ) : <PotentialLocations />
          }
        </div>
      </main>
    )
  }
}


export const mapStateToProps = state => ({
  city: state.city,
  allTrucks: state.allTrucks
})

export const mapDispatchToProps = dispatch => ({
  setTrucks: trucks => dispatch(setTrucks(trucks)),
  togglePotentialLocation: truck => dispatch(togglePotentialLocation(truck))
})

export default connect(mapStateToProps, mapDispatchToProps)(TrucksContainer);