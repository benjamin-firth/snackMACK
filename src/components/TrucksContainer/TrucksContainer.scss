.right-side-box {
  width: 100%;
}

.leaflet-container {
  width: 100%;
  height: 100vh;
}

.mainpage-container {
  background-color: #231E42;
  display: flex;
}

.sidebar {
  background-color: #231E42;
  border-right: 2px solid #83A1B0;
  padding: 20px;
  width: 40%;

  button {
    background-color: #83A1B0;
    border-radius: 5px;
    color: #302D50;
    font-size: 21px;
    margin-bottom: 25px;
    width: 100%;
  }
  
  .inner-scroll-container {
    overflow: scroll;
    height: 83vh;
  }
}