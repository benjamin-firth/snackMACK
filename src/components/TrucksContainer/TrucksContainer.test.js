import React from 'react';
import { shallow } from 'enzyme';
import { TrucksContainer } from './TrucksContainer';

describe('TrucksContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <TrucksContainer />)
  })

it('should match the snapshot', () => {
    // assert.isDefined(AccountFormComponent);
    expect(wrapper).toMatchSnapshot();
  })

})