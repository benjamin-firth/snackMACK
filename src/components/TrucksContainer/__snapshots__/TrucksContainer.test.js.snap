// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrucksContainer should match the snapshot 1`] = `
<main>
  <Header />
  <div
    className="mainpage-container"
  >
    <div
      className="sidebar"
    >
      <button>
        VIEW POTENTIAL LOCATIONS
      </button>
      <div
        className="inner-scroll-container"
      />
    </div>
    <Connect(PotentialLocations) />
  </div>
</main>
`;
